### PR TITLE
Refactor/reducer

### PR DESCRIPTION
**Pull request by zhouyuhe**
Split our reducer to components reducer

### CODE REVIEW

**Comment by Sharakai** [comment]
Looking at the commit messages, this should have been around 5 different PRs:
1. "Move all components inside features folder"
2. "Split up reducer for all referenceData" & "Split reducers to each components"
3. "Add Jest accordingly"
4. "fix key stats ytd change"
5. "change color for axis"

Not splitting these up now means reviewers have to review 5 large changes, across 58 files, all at once. It's rare I will ever bother doing that, unless it is absolutely necessary, leading to people's PRs not being merged in for days and days and it essentially becoming "stale" and out of date.

File: src/Reducer.test.js [hunk 1]
```diff
@@ -15,18 +15,30 @@ import {
```
Comment (Sharakai) [inline src/Reducer.test.js]: This test file should be split out in the same way you've split out your reducers. Each reducer ~can be~ should be tested independently.

File: src/features/chart/components/Chart.js [hunk 2]
```diff
@@ -11,9 +11,9 @@ import {
 } from "recharts";
 // import moment from 'moment';
 import { useSelector, useDispatch } from "react-redux";
-import { updateChartAction } from "../../store/actions";
-import { Loading } from "../loading";
-import { ErrorMessage } from "../error-message";
+import { updateChartAction } from "../redux/actions";
+import { Loading } from "../../loading/components/Loading";
```
Comment (Sharakai) [inline src/features/chart/components/Chart.js]: nit: This import shouldn't have to change. The component should be exported from an `index.js` file in the `loading` feature folder.

Same applies to every other feature folder, by the way. You should consider what a feature's "API" consists of:
1. Component
2. Reducer
3. Middleware
4. Actions

File: src/Reducer.test.js (hunk 1 above)
Comment (zhouyuhe) [reply src/Reducer.test.js]: we'll do it in another branch

File: src/features/chart/components/Chart.js (hunk 2 above)
Comment (steinalex) [reply src/features/chart/components/Chart.js]: We'll change this in another refactor branch

**Pull request merged**